### PR TITLE
Misc. items

### PR DESCRIPTION
**Pull request by brettz9**
I have a number of smaller items offered for adding here, but I thought I'd just submit this and see if you had preferences off the bat for grouping (or not accepting some of the items) before I might do any breaking of it into smaller chunks. You can see the individual commits though to note the atomic changes.

### CODE REVIEW

**Comment by brettz9** [comment]
Note that while CI is failing due to ESLint@6 no longer working with Node@6, I can either add logic to have the Node 6 test use ESLint 5 or just put Node 8 as the minimum supported version.

**Comment by bergos** [comment]
Thanks! I like almost all of it, besides:

- replacing `standard` with `eslint` + config and removing the globals in the test files. I like to keep `standard` so there is not too much boilerplate code + configs. The globals definition for the test functions makes it more readable for people that don't know the complete js ecosystem.
- I don't remember in detail, but I think there was a problem when `package-lock.json` was added to `.gitignore`. I think npm complained about it when I tried to publish the package.
- Why did you rearrange the requires? I ask out of curiosity. My current logic: Sorting by package name, functions before classes. Because of ES6 destructors, I gave up functions vs. classes and use only package names.

About the travis config and supported Node versions: In the more generic libraries like this one, I try to support all LTS versions, so this would be 8, 10, 12 at the moment and 6 can be removed from the travis config.

**Comment by brettz9** [comment]
Regarding `standard` configs, yeah, I understand both about not wanting the extra boilerplate and the unexplained "magic" of `describe`. Re: the former, it's a pity there is no succinct way to get peers tacked on for (dev) users automatically. Re: the latter, I believe if using ESM at least, one can use `import {describe} from 'mocha';`. The cost at which this comes of developers not being able to see linting errors until testing is a real trade-off for rapid development though. But in this case, it's a small library at least.

Re: `package-lock.json`, are you ok my leaving the change and just trying for publishing? I have done this libraries I work on, and it works ok. Maybe that was for an older version of npm?

Re: rearranging the requires, I'm following the convention I've found helpful from https://github.com/benmosher/eslint-plugin-import/blob/master/docs/rules/order.md . By putting the builtins first (and third-party next before internal requires/imports), I find it has these advantages:

1. I can selectively ignore the Node builtins to just see what third-party dependencies there are (e.g., to determine if any aren't really necessary).
2. I can quickly take note of the Node builtins in use (esp. comprehensive when other rules are present preventing `global-require`) to know which fundamental features/permissions are required. Looking ahead, I'm expecting this could assist with any migrations from Node to [Deno](https://deno.land/std/manual.md), a platform which requires opting in to special permissions like file system or network access; grouping the fundamental libraries could help one more quickly identify those which are in use.

Let me know if you want me to revert. I'll just make the Travis and eslint config changes for now.

**Comment by brettz9** [comment]
I've made the Travis change and eslint config changes, though I added a commit (which I can revert if you prefer) to switch from the `/* global */` directive to `/* eslint-env mocha */` which I think allows users to find out what "eslint-env" is, conveys the reasons (part of the mocha environment), as well as future-proofs it if you need to use other mocha globals. But again, let me know if you want me to revert that or the require sorting order, etc.

**Comment by brettz9** [comment]
Oh, and I noticed there was a "standard" update which I've also added (no changes needed).

**Comment by bergos** [comment]
Just tried `const { describe, it } = require('mocha')` and it works! Didn't think of that cause I tried that recently with `jest` and it didn't work. I think that's the way to go.

Let's try if the `.gitignore` works now with `package-lock.json`.

The order of the requires also makes sense to me. So let's keep that.

**Comment by brettz9** [comment]
Ok, applied! See what you think / if it works...

**Comment by brettz9** [comment]
I know it's only been 4 days, but just wondering if you might know when you'd expect to take a look? Thanks :)

File: package.json [hunk 1]
```diff
@@ -18,19 +18,25 @@
...
   "author": "Thomas Bergwinkl <bergi@axolotlfarm.org> (https://www.bergnet.org/people/bergi/card#me)",
+  "contributors": [
+    "Brett Zamir"
+  ],
   "license": "MIT",
   "bugs": {
     "url": "https://github.com/bergos/file-fetch/issues"
   },
   "homepage": "https://github.com/bergos/file-fetch",
+  "engines": {
```
Comment (bergos) [inline package.json]: It should be aligned with the Travis config or maybe we don't add it at all. I would prefer the last option. I never checked in detail what effect the `engines` property has. I think Node would throw an error, if the version doesn't match. As the compatible versions are sync with the LTS versions, I see it as the users responsibility to handle problems caused by unsupported Node versions.

Comment (brettz9) [reply package.json]: `engines` only has an impact if the user opts in to the config `engine-strict`, and then will err out only if the package is added as a dependency.

Besides flagging for users whether they may continue using an older (or later) Node version, this field has been used by linting tools to draw attention to whether one mistakenly introduces syntax or features which are outside of the advertised `engines` range. But since the rules of "standard" do not make use of this (despite being potentially available through one of their peer deps, `eslint-plugin-node`, e.g., see https://github.com/mysticatea/eslint-plugin-node/tree/master/docs/rules/no-unsupported-features , and even if you were using this, it defaults to LTS versions anyways), this latter advantage appears immaterial in this case.

I've updated to drop `engines`.

**Comment by bergos** [comment]
@brettz9 just published version 1.3.0

**Comment by brettz9** [comment]
Thanks! Quite happy to have full parity with browser tests... :)
